.github/workflows/reusable_publish_image.yml

```diff
@@ -0,0 +1,246 @@
+# Note: Repository secret scanning and Dockerfile linting are handled by reusable_ci.yml (MegaLinter)
+
+name: Build and Publish Image (Reusable)
+
+on:
+  workflow_call:
+    inputs:
+      allowed_identity_regex:
+        description: 'Regex for Cosign certificate identity (optional, defaults to org-level)'
+        required: false
+        type: string
+      component_name:
+        description: 'Component name (e.g., compass, beacon-distro)'
+        required: true
+        type: string
+      context_path:
+        description: 'Build context path (e.g., ./compass)'
+        required: true
+        type: string
+      dockerfile_path:
+        description: 'Dockerfile path relative to repo root (e.g., compass/images/Containerfile.compass)'
+        required: true
+        type: string
+      image_name:
+        description: 'Full image name (e.g., ghcr.io/complytime/complybeacon-compass)'
+        required: true
+        type: string
+      force_rebuild:
+        description: 'Force rebuild without cache'
+        required: false
+        type: boolean
+        default: false
+      dry_run:
+        description: 'Run security checks without pushing or signing'
+        required: false
+        type: boolean
+        default: false
+
+permissions:
+  contents: read
+
+concurrency:
+  group: ${{ github.workflow }}-${{ inputs.image_name }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build-and-push:
+    runs-on: ubuntu-latest
+    timeout-minutes: 60
+    permissions:
+      contents: read
+      packages: write
+      id-token: write        # for OIDC (attestations / cosign keyless)
+      security-events: write # for the Trivy SARIF step
+    steps:
+      - uses: actions/checkout@8e8c483db84b4bee98b60c0593521ed34d9990e8 # v6.0.1
+
+      - name: Docker metadata
+        id: meta
+        uses: docker/metadata-action@c299e40c65443455700f0fdfc63efafe5b349051 # v5.10.0
+        with:
+          images: ${{ inputs.image_name }}
+          tags: |
+            type=semver,pattern={{version}}
+            type=semver,pattern={{major}}.{{minor}}
+            type=semver,pattern={{major}},enable=${{ !startsWith(github.ref, 'refs/tags/v0.') }}
+            type=ref,event=branch,suffix=-{{sha}}
+            type=sha,prefix=sha-
+            type=schedule,pattern={{date 'YYYYMMDD-HHmmss'}},prefix=scheduled-
+            type=raw,value=latest,enable={{is_default_branch}}
+          flavor: |
+            latest=auto
+          labels: |
+            org.opencontainers.image.title=${{ inputs.component_name }}
+            org.opencontainers.image.description=ComplyBeacon ${{ inputs.component_name }} service
+            org.opencontainers.image.vendor=ComplyTime
+
+      - name: Log in to GHCR
+        uses: docker/login-action@5e57cd118135c172c3672efd75eb46360885c0ef # v3.6.0
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
Do you need to pass in this secret or would this expect to be used with secrets inherited?
 ```
secrets:
      token:
        required: true
```